dhall/src/Dhall/TypeCheck.hs

```diff
@@ -3041,7 +3046,17 @@ prettyTypeMessage (MustMergeUnion expr0 expr1) = ErrorMessages {..}
         \    └─────────────────────────────────────────────────────────────────┘         \n\
         \                                                                                \n\
         \                                                                                \n\
-        \... but the second argument to ❰merge❱ must be a union and not some other type. \n\
+        \... or this:                                                                    \n\
+        \                                                                                \n\
+        \                                                                                \n\
+        \    ┌─────────────────────────────────────────────────────────┐                 \n\
+        \    │ let optional = None Bool                                │                 \n\
+        \    │ let handlers = { None = False, Some = λ(x : Bool) → x } │                 \n\
+        \    │ in  merge handlers optional : Bool                      │                 \n\
```

Comment:
Is the type annotation required?

```suggestion
        \    │ in  merge handlers optional                             │                 \n\
```

Reply:
I have removed the type annotation in some other error messages for `merge` too.